Banking_App/src/main/java/com/cognologix/assignment/BankSystemAssignment/services/impl/TransactionServiceImplementation.java

```diff
@@ -0,0 +1,176 @@
+        Transaction transaction = new Transaction();
+
+        //checking account is exit or not in account table
+        Account account = this.accountRepo.findById(accountNumber)
+                .orElseThrow(()->new ResourceNotFoundException("Account","id",accountNumber));
+
+
+        Double totalBalance = account.getAccountBalance();
+
+        if(totalBalance>=withdrawAmount){
+            transaction.setTransferAmount(withdrawAmount);
+            //transaction.setDepositAmount(0.0);
+            transaction.setSenderAccountNumber(accountNumber);
+            Double balanceAfterWithdrawl = totalBalance-withdrawAmount;
+            System.out.println("Balance After withdrawal is " +balanceAfterWithdrawl);
+
+            transaction.setTotalAmount(balanceAfterWithdrawl);
+            account.setAccountBalance(balanceAfterWithdrawl);
+
+            this.transactionRepo.save(transaction);
+            this.accountRepo.save(account);
+        }else {
+            throw new InsufficientBalanceException("Your Account Balance is Low");
+        }
+
+    }
+
+    /*
+    * Transfer amount from one account to another
+    * */
+
+    @Override
+    public void amountTransfer(Integer senderAccountNumber, Integer receiverAccountNumber, Double transferAmount) {
+//       getDepositAmount(receiverAccountNumber,transferAmount);
+//       getWithDrawAmount(senderAccountNumber,transferAmount);
+        /*
+        *   sender
+        */
+       // Transaction senderTransaction = this.transactionRepo.findByAccountNumber(senderAccountNumber);
+
+        Transaction transaction = new Transaction();
+
+        //checking account is exit or not in account table
+        Account senderAccount = this.accountRepo.findById(senderAccountNumber)
+                .orElseThrow(()->new ResourceNotFoundException("Account","id",senderAccountNumber));
+
+        //finding total balance
+        Double totalBalance = senderAccount.getAccountBalance();
+
+        if(totalBalance>=transferAmount){
+            Double balanceAfterWithdrawl = totalBalance-transferAmount;
+            System.out.println("Balance After withdrawal is " +balanceAfterWithdrawl);
+            transaction.setSenderAccountNumber(senderAccountNumber);
+            transaction.setTransferAmount(transferAmount);
+            transaction.setTotalAmount(balanceAfterWithdrawl);
+            this.transactionRepo.save(transaction);
+            this.accountRepo.save(senderAccount);
+        }else {
+            throw new InsufficientBalanceException("Your Account Balance is Low");
+        }
+
+        /*
+        *   ---- Receiver
+        * */
+       // Transaction receiverTransaction = this.transactionRepo.findByAccountNumber(receiverAccountNumber);
+
+        //checking account is exit or not in account table
+        Account receiverAccount = this.accountRepo.findById(receiverAccountNumber)
+                .orElseThrow(()->new ResourceNotFoundException("Account","id",receiverAccountNumber));
+
+
```

Comment:
check sender and receiver both exist and the check for balance and then do the transaction